test/integration/volume/persistent_volumes_test.go

```diff
@@ -1068,7 +1068,11 @@ func TestPersistentVolumeProvisionMultiPVCs(t *testing.T) {
 		}
 		for {
 			select {
-			case event := <-w.ResultChan():
+			case event, ok := <-w.ResultChan():
+				if !ok {
+					t.Error("watch channel should not close")
```

Comment:
probably want to just log here, not error, unless we're wanting to make the test fail in new ways

a watch closing is a normal thing, not an error

Reply:
alternatively, you could log this happened and restart the watch

Reply:
Is it like this? @liggitt 

```go
		for {
			select {
			case event, ok := <-w.ResultChan():
				if !ok {
					klog.Info("Event watch channel closed")
					w, err = testClient.EventsV1().Events(ns.Name).Watch(tCtx, metav1.ListOptions{})
					if err != nil {
						return
					}
					continue
				}
				reportToArtifacts(t.Name()+"-events.text", event.Object)
			case <-tCtx.Done():
				w.Stop()
				return
			}
		}
```

Reply:
I'm not sure if we need to restart the watch

Reply:
yeah, that looks reasonable if we're just debugging

Reply:
updated